src/main/scala/mesosphere/marathon/Main.scala

```diff
@@ -27,7 +30,32 @@ class MarathonApp extends App {
     val client = new ZooKeeperClient(
       Amount.of(conf.zooKeeperSessionTimeout().toInt, Time.MILLISECONDS),
       conf.zooKeeperHostAddresses.asJavaCollection
-    )
+    ) {
+      override def shouldRetry(e: KeeperException): Boolean = {
+        log.error(s"Got ZooKeeper exception: $e")
+        log.error("Committing suicide to avoid invalidating ZooKeeper state")
```

Comment:
I would add a comment which explains the code.
* Why is it important to stop the process?
* What is the problem with `exit`?